### PR TITLE
MysqlDataTruncation: Data truncation: Data too long for column 'metho…

### PR DESCRIPTION
**Pull request by azorinmsu**
…d' #398

---
This change is [Reviewable](https://reviewable.io/reviews/php-coder/mystamps/416)

### CODE REVIEW

**Comment by php-coder** [comment]
Please, don't include the issue number in the commit description. Because this commit will be modified multiple times we don't want to have many references to old commits in the issue.

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[src/main/java/ru/mystamps/web/Db.java, line 38 \[r1\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KN8443O5A3TrwG72Nu9:-KN8443PGRGKCs0egQDv:-1199405248) ([raw file](https://github.com/php-coder/mystamps/blob/1a922be6c0e51beb2f3cd71e079de59a18c4ebda/src/main/java/ru/mystamps/web/Db.java#L38)):*
> ```Java
> 		public static final int REFERER_PAGE_LENGTH = 255;
> 		public static final int USER_AGENT_LENGTH   = 255;
> 		public static final int METHOD_LENGTH = 8;
> ```

1) align this line with spaces (like other)
2) move this line above, right after `PAGE_URL_LENGTH`

---

*[src/main/java/ru/mystamps/web/service/SiteServiceImpl.java, line 156 \[r1\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KN851J4KbWUT8OpKSht:-KN851J4KbWUT8OpKShu:1533979400) ([raw file](https://github.com/php-coder/mystamps/blob/1a922be6c0e51beb2f3cd71e079de59a18c4ebda/src/main/java/ru/mystamps/web/service/SiteServiceImpl.java#L156)):*
> ```Java
> 	}
> 	
> 	private static String abbreviateMethod(String method){
> ```

When you're adding a new method you should put the comment with info about author:
```java
/**
  * @author You Name
  */
```

Ideally you can also add description of the method and info about parameters/return value/exceptions, but it's not required and up to you.

---

*[src/main/resources/liquibase/version/0.4.xml, line 17 \[r1\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KN84ibKrQQSIgI4mFDF:-KN84ibKrQQSIgI4mFDG:1163275537) ([raw file](https://github.com/php-coder/mystamps/blob/1a922be6c0e51beb2f3cd71e079de59a18c4ebda/src/main/resources/liquibase/version/0.4.xml#L17)):*
> ```XML
> 	<include file="0.4/2016-02-19--csrf_events.xml" relativeToChangelogFile="true" />
> 	<include file="0.4/2016-01-14--unique_slug_in_countries.xml" relativeToChangelogFile="true" />
> 	<include file="0.4/2016-07-20--change-length-field-method.xml" relativeToChangelogFile="true" />
> ```

Unfortunately you did a bit of extra work that we don't need. We don't need to increase the size of the column.

---

*[src/main/resources/liquibase/version/0.4/2016-07-20--change-length-field-method.xml, line 13 \[r1\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KN84fjK9GEezXwn1Sc0:-KN84fjK9GEezXwn1Sc1:-1306927275) ([raw file](https://github.com/php-coder/mystamps/blob/1a922be6c0e51beb2f3cd71e079de59a18c4ebda/src/main/resources/liquibase/version/0.4/2016-07-20--change-length-field-method.xml#L13)):*
> ```XML
>   </changeSet>
> 
> </databaseChangeLog>
> ```

We don't need this migration.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/php-coder/mystamps/416#-:-KN83Pkv99NkqkgoE6nQ:-624088719)*

**Comment by php-coder** [comment]
Currently some of the checks failed in the TravisCI. Please, fix error from CheckStyle.

---

Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy, line 347 \[r1\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KN87UCkcMypBN9lTaEH:-KN87UCl6OdzAEReF9_T:-609915689) ([raw file](https://github.com/php-coder/mystamps/blob/1a922be6c0e51beb2f3cd71e079de59a18c4ebda/src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy#L347)):*
> ```Groovy
> 				return true
> 			})
> 	}
> ```

I'm trying to order the tests in the same order as the code lines in the original method. So, tests for setting the method should be not the last, but after the "logEvent() should pass method to dao" method.

After moving this method to the proper place, we'll see than we already have similar test. And it means that we can just add one more line to "logEvent() should pass method to dao" method.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/php-coder/mystamps/416#-:-KN86ziYtOnTcxXa2-pS:-803213130)*

**Comment by azorinmsu** [comment]
Thank you for review all defects was repaired

**Comment by php-coder** [comment]
Ok, thank you! Need a bit more polishing.

---

Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[src/main/java/ru/mystamps/web/Db.java, line 35 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfsE3maq9yupfPkkyk:-KNfsE3maq9yupfPkkyl:-1393239609) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/main/java/ru/mystamps/web/Db.java#L35)):*
It's better now, but I still don't see the changes :)

> 1) align this line with spaces (like other)

For aligning inside of the line we're using spaces.

> 2) move this line above, right after PAGE_URL_LENGTH

Pay attention: `right after` != `right before`

---

*[src/main/java/ru/mystamps/web/service/SiteServiceImpl.java, line 155 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfpwHqMCExNYbUAdsy:-KNfpwHqMCExNYbUAdsz:-1666190110) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/main/java/ru/mystamps/web/service/SiteServiceImpl.java#L155)):*
> ```Java
> 		suspiciousActivities.add(activity);
> 	}
> ```

Try to not modify code that not needed for you change. In this case, I would ask you to put this tab back and reduce the diff.

---

*[src/main/java/ru/mystamps/web/service/SiteServiceImpl.java, line 156 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfrOKpnzTRYOv3viQB:-KNfrOKpnzTRYOv3viQC:1306419049) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/main/java/ru/mystamps/web/service/SiteServiceImpl.java#L156)):*
> ```Java
> 	}
> 
> 	/**
> ```

I like your idea of adding javadoc comments. To make it "real" we need also add a small description for this method :)

BTW, don't forget that actually javadoc isn't required for private methods, like this.

---

*[src/main/java/ru/mystamps/web/service/SiteServiceImpl.java, line 157 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfri-9ny319IEYKsdv:-KNfri-9ny319IEYKsdw:473942632) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/main/java/ru/mystamps/web/service/SiteServiceImpl.java#L157)):*
> ```Java
> 
> 	/**
> 	 * @param method name of method's http
> ```

`@param method name of HTTP method`

---

*[src/main/java/ru/mystamps/web/service/SiteServiceImpl.java, line 158 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfrS4uIyxGfwYgaEUc:-KNfrS4uIyxGfwYgaEUd:803893702) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/main/java/ru/mystamps/web/service/SiteServiceImpl.java#L158)):*
> ```Java
> 	/**
> 	 * @param method name of method's http
> 	 * @return abbreviated name of method with three point at the end
> ```

Let's add more details and improve English a bit (three point -> three points): `@return name of the method as-is or its abbreviation with three points at the end`

---

*[src/main/java/ru/mystamps/web/service/SiteServiceImpl.java, line 160 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfr08B1lnWPMOYNmfm:-KNfr08B1lnWPMOYNmfn:168034744) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/main/java/ru/mystamps/web/service/SiteServiceImpl.java#L160)):*
> ```Java
> 	 * @return abbreviated name of method with three point at the end
> 	 * @author Aleksandr Zorin
>    */
> ```

End of the comment aligned with spaces.

---

*[src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy, line 205 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfofZM3gBBFsDHSr9f:-KNfofZM3gBBFsDHSr9g:1115955173) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy#L205)):*
> ```Groovy
> 	}
> 
> 	def "logEvent() should abbreviate method when pass argument greater than 7 symbols"() {
> ```

Let's name it like other similar tests: `logEvent() should pass abbreviated method when it's too long`. Benefits: 1) all tests use the same naming strategy 2) we didn't have too much details in the description. It means that when we change the max length of the method to 8, we won't need to update test name.

---

*[src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy, line 216 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfotOlV2Zxvx_ht0gY:-KNfotOlV2Zxvx_ht0gZ:833382167) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy#L216)):*
> ```Groovy
> 			assert activity?.method == exceptedMethod
> 			return true
> 		})
> ```

Please, format the code in the same style as other tests. For example:

```
given:
    String method = "PROPFIND"
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/php-coder/mystamps/416#-:-KNfsbTV-QEWdKjl_SQH:-2090650241)*

**Comment by php-coder** [comment]
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/main/java/ru/mystamps/web/service/SiteServiceImpl.java, line 155 \[r2\]](https://reviewable.io:443/reviews/php-coder/mystamps/416#-KNfpwHqMCExNYbUAdsy:-KO9bzjUMzdqV3jocMyM:1268096115) ([raw file](https://github.com/php-coder/mystamps/blob/c0a03342f3c0d489ac254ce2e14d2217228f0105/src/main/java/ru/mystamps/web/service/SiteServiceImpl.java#L155)):*
<details><summary><i>Previously, php-coder (Vyacheslav Semushin) wrote…</i></summary>
> Try to not modify code that not needed for you change. In this case, I would ask you to put this tab back and reduce the diff.
</details>
Still here. Please put newline back.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/php-coder/mystamps/416)*

File: src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy [hunk 1]
```diff
@@ -201,6 +201,20 @@ class SiteServiceImplTest extends Specification {
...
+			String method = "PROPFIND"
+		and:
+			String exceptedMethod = method.take(Db.SuspiciousActivity.METHOD_LENGTH-3) + '...'
+		when:
+			serviceImpl.logEvent(TEST_TYPE, TEST_PAGE, method, null, null, null, TEST_USER_AGENT, null)
+		then:
+			1 * suspiciousActivityDao.add({ AddSuspiciousActivityDbDto activity ->
+				assert activity?.method == exceptedMethod
+				return true
+		})
```
Comment (php-coder) [inline src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy]: Could you add a one tab to this line? :)

Comment (azorinmsu) [inline src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy]: @php-coder  Which line have i need add tab 216 or 217 ?

Comment (php-coder) [inline src/test/groovy/ru/mystamps/web/service/SiteServiceImplTest.groovy]: 216 (just look at other tests to see an example of how it should be formatted).

**Comment by php-coder** [comment]
Fixed in 1f571a5ba325e6b34b15f67136998be086598c5a commit.

**Pull request closed without merging**